sys/shell/commands/sc_ccnl.c

```diff
@@ -89,37 +89,35 @@ static void _content_usage(char *argv)
 
 int _ccnl_content(int argc, char **argv)
 {
-    char *body = (char*) _default_content;
-    int arg_len = strlen(_default_content) + 1;
-    int offs = CCNL_MAX_PACKET_SIZE;
     if (argc < 2) {
         _content_usage(argv[0]);
         return -1;
     }
 
+    int arg_len;
+    char *body = (char*) _default_content;
+    char buf[BUF_SIZE];
+
     if (argc > 2) {
-        char buf[BUF_SIZE];
-        char *buf_ptr = buf;
-        for (int i = 2; (i < argc) && (buf_ptr < (buf + BUF_SIZE)); i++) {
-            if (i > 2) {
-                *(buf_ptr++) = ' ';
-            }
+        unsigned pos = 0;
+        for (int i = 2; (i < argc) && (pos < BUF_SIZE); ++i) {
             arg_len = strlen(argv[i]);
-            if ((buf_ptr + arg_len) > (buf + BUF_SIZE)) {
-                arg_len = (buf + BUF_SIZE) - buf_ptr;
+            if ((pos + arg_len) > BUF_SIZE) {
+                arg_len = BUF_SIZE - pos;
             }
-            strncpy(buf_ptr, argv[i], arg_len);
-            buf_ptr += arg_len;
+            strncpy(&buf[pos], argv[i], arg_len);
+            pos += arg_len;
+            /* increment pos _after_ adding ' ' */
+            buf[pos++] = ' ';
```

Comment:
btw. surpassing the buffer size with `pos++` would be fine (and actually is expected), cause it increments _after_ using `pos`. However, with `pos += arg_len` it may already point beyond the actual buffer.